UserDetailsClient/UserDetailsClient.Core/MainPage.xaml.cs

```diff
@@ -45,15 +24,19 @@ public MainPage()
                 {
                     AuthenticationResult ar = await App.PCA.AcquireTokenAsync(App.Scopes, GetUserByPolicy(App.PCA.Users, App.PolicySignUpSignIn), App.UiParent);
                     UpdateUserInfo(ar);
+                    Device.BeginInvokeOnMainThread(() => { btnSignInSignOut.Text = "Sign out"; });
                     UpdateSignInState(true);
+
```

Comment:
Addressing threading issues here, like the [v2-xamarin app](https://github.com/Azure-Samples/active-directory-xamarin-native-v2/blob/master/UserDetailsClient/UserDetailsClient/MainPage.xaml.cs#L32).